lib/veres-one-client.js

```diff
@@ -0,0 +1,187 @@
+
+   * @param [mode='test'] {string} One of 'dev'/'test'/'live'. Determines https
+   *   agent settings (does not reject unsigned certs in 'dev' mode, for
+   *   example).
+   * @param [logger]
+   * @param [httpsAgent] {Agent}
+   */
+  constructor({injector, hostname, ledger, mode, logger, httpsAgent}) {
+    this.injector = injector;
+    this.hostname = hostname;
+    if(!hostname) {
+      throw new Error('Missing ledger hostname');
+    }
+    this.ledger = ledger ||
+      new WebLedgerClient({injector, hostname, mode, logger, httpsAgent});
+
+    this.mode = mode || DEFAULT_MODE;
+    this.logger = logger || console;
+
+    this.httpsAgent = httpsAgent;
+    if(mode === 'dev' && !this.httpsAgent) {
+      this.httpsAgent = new https.Agent({rejectUnauthorized: false});
+    }
+  }
+
+  /**
+   * Fetches a DID Document for a given DID. If it contains a #hash fragment,
+   * it's likely a key id, so just return the subgraph, not the full doc.
+   *
+   * @param did {string} DID uri (possibly with hash fragment)
+   *
+   * @returns {Promise<object>} Resolves to DID Document Fetch Result
+   */
+  async get({did}) {
+    if(!did) {
+      throw new Error('Invalid or missing DID URI');
+    }
+    const [docUri, hashFragment] = did.split('#');
+
+    const result = await this.ledger.get({id: docUri});
+
+    const didDoc = result.record;
+
+    const context = didDoc['@context'];
+
+    if(!hashFragment) {
+      // full DID Doc
+      result.type = 'LedgerDidDocument';
+      result.doc = didDoc;
+    } else {
+      // Request for a subgraph (likely just the key node)
+      const jsonld = this.injector.use('jsonld');
+      const map = await jsonld.createNodeMap(didDoc);
+      const subGraph = map[did];
+      if(!subGraph) {
+        throw new Error(
+          `Failed to get subgraph within a DID Document, uri: ${did}`
+        );
+      }
+
+      // result.type = 'Key'; <- not sure what this should be
+      result.doc = await jsonld.compact(subGraph, context);
+    }
+
+    return result;
+  }
+
+  /**
+   * Sends an operation to a Veres One accelerator.
+   *
+   * @param options {object}
+   *
+   * @param options.operation {object} WebLedgerOperation
+   *
+   * @param [options.hostname] {string} Accelerator hostname
+   * @param [options.env] {string} Used to determine default hostname
+   *
+   * Keys for signing the http request headers
+   * @param [options.authKey] {LDKeyPair}
+   *
+   * @returns response {Promise<object>} from an axios POST request
+   */
+  async sendToAccelerator(options) {
+    const {operation, authKey} = options;
+    const hostname = options.hostname || this.hostname;
+
+    const acceleratorPath = '/accelerator/proofs';
+    const acceleratorUrl = `https://${hostname}${acceleratorPath}`;
+
+    const headers = {
+      'Accept': 'application/ld+json, application/json',
+      'Host': hostname
+    };
+
+    if(authKey && authKey.keyType === 'RsaVerificationKey2018') {
```

Comment:
What's going on here? Are we only supporting `RsaVerificationKey2018`? Why are we only signing if that's the key type? What's supposed to happen if that's *not* the key type? It looks like `privateKeyBase58` is used below ... which only happens when the `keyType` is `Ed25519VerificationKey2018`.

Reply:
All fine questions. I do not know, I just moved the existing function here. I'm happy to remove that guard statement.

Reply:
Fixed.

Reply:
Er, wait. Ok, so I have two questions.

1. Does the http signatures library support ed25519 keys? (the version I'm currently using does not seem to.)
2. Should we require a key for sending to accelerator? (Or are unsigned requests possible?)

Reply:
Hmm, you might be using the wrong version of the lib? Should be using something like this:  https://github.com/digitalbazaar/http-signature-header

Reply:
cc: @mattcollier

Reply:
@dmitrizagidulin yes, if you want to do httpsignature with ed25519, you must use the library longley mentions above.  See this test helper for how to use this module: https://github.com/digitalbazaar/http-signature-middleware/blob/master/test/helpers.js#L14

@dlongley that said, I'm confused because I did not think we were using httpsignatures for veres-one.  This is for submitting operations (did-documents) right?  There is no authz on the operations endpoint: https://github.com/digitalbazaar/bedrock-ledger-agent/blob/master/lib/http.js#L281

I thought authz was being handled by verification of signatures/capabilities embedded in the operation.

Reply:
@mattcollier, this is for accelerators. And now I don't remember how we do authz on those anyway... but I think you have to establish an account and then use http-signatures.

Reply:
Ok, so, I'll change the code to be the same as that test helper @mattcollier linked to above.

Question, though: Is there any way we can extract or re-use that method? Instead of copying it in at least 2 repos?